providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java

```diff
@@ -1,29 +1,294 @@
 package dev.openfeature.contrib.providers.flagd;
 
-import dev.openfeature.javasdk.Client;
-import dev.openfeature.javasdk.NoOpProvider;
-import dev.openfeature.javasdk.OpenFeatureAPI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
-/** 
- * A placeholder.
+import org.apache.commons.lang3.EnumUtils;
+
+import dev.openfeature.flagd.grpc.Schema.ResolveBooleanRequest;
+import dev.openfeature.flagd.grpc.Schema.ResolveBooleanResponse;
+import dev.openfeature.flagd.grpc.Schema.ResolveFloatRequest;
+import dev.openfeature.flagd.grpc.Schema.ResolveFloatResponse;
+import dev.openfeature.flagd.grpc.Schema.ResolveIntRequest;
+import dev.openfeature.flagd.grpc.Schema.ResolveIntResponse;
+import dev.openfeature.flagd.grpc.Schema.ResolveObjectRequest;
+import dev.openfeature.flagd.grpc.Schema.ResolveObjectResponse;
+import dev.openfeature.flagd.grpc.Schema.ResolveStringRequest;
+import dev.openfeature.flagd.grpc.Schema.ResolveStringResponse;
+import dev.openfeature.flagd.grpc.ServiceGrpc;
+import dev.openfeature.flagd.grpc.ServiceGrpc.ServiceBlockingStub;
+import dev.openfeature.javasdk.EvaluationContext;
+import dev.openfeature.javasdk.FeatureProvider;
+import dev.openfeature.javasdk.Metadata;
+import dev.openfeature.javasdk.ProviderEvaluation;
+import dev.openfeature.javasdk.Reason;
+import dev.openfeature.javasdk.Structure;
+import dev.openfeature.javasdk.Value;
+import io.grpc.ManagedChannelBuilder;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * OpenFeature provider for flagd.
  */
-public class FlagdProvider {
+@Slf4j
+public class FlagdProvider implements FeatureProvider {
+
+    private ServiceBlockingStub serviceStub;
+    static final String PROVIDER_NAME = "flagD Provider";
+
+    /**
+     * Create a new FlagdProvider instance.
+     *
+     * @param protocol transport protocol, "http" or "https"
+     * @param host     flagd host, defaults to localhost
+     * @param port     flagd port, defaults to 8013
+     */
+    public FlagdProvider(String protocol, String host, int port) {
```

Comment:
Can protocol be an enum and can port be a short?

Reply:
> Can protocol be an enum and can port be a short?

`short` goes up to 32767, while ports go up to 65535, so I think that's a bad idea. I can make the protocol an enum.

Reply:
Done!